### DISCUSSION TITLE
Vulnerability scans

### DISCUSSION THREAD

**Discussion by SandroCristiano** [wazuh/wazuh#13344] [Q&A] [answered]
Hello! I'm having difficulties to understand how Vulnerability scans work. I configured everything as per the documentation but it still not happening. Also, isn't there a more intuitive way do it, like a simple "Vuln Scan" button? Thanks.

**Comment by HanesSciarrone** [comment]
Hi @SandroCristiano 

I hope you are well, thanks for using Wazuh and sharing your question with the community. Well, regarding having a more intuitive way to generate a vulnerability scan currently the development team at Wazuh is working to provide a way to generate a vulnerability scan on-demand in addition to time intervals. If you want to follow the development you can see this [epic](https://github.com/wazuh/wazuh/issues/9220). Please note that this Epic only involves the backend functionality for the UI there should be another issue with the front-end team that hasn't yet been created I think.

Regarding the vulnerability detector configuration, I would like to know if you have followed these steps:

1. In the **ossec.conf** of the manager side enable the Vulnerability detector, by default it is disabled.

``` xml
<vulnerability-detector>
  <enabled>yes</enabled>
  <interval>5m/interval>
  <run_on_start>yes</run_on_start>
</vulnerability-detector>
```

2. The provider tags should be placed depending on the OS where the agents are running. For example for Ubuntu is:

``` xml
  <provider name="canonical">
    <enabled>yes</enabled>
    <os>bionic</os>
    <update_interval>1h</update_interval>
  </provider>
```

3. On the agent side Syscollector must be enabled and with the packages and hotfixes tag set to yes (note the hotfixes is only for Windows). I leave you an example

``` xml
<wodle name="syscollector">
  <disabled>no</disabled>
  <interval>20m</interval>
  <os>yes</os>
  <packages>yes</packages>
  <hotfixes>yes</hotfixes>
</wodle>
```
4. Agents should be active and directly connected to the manager where the vulnerability detector run.

Best regards
Hanes

**Reply by SandroCristiano** [reply]
Thank you Hanes. It worked as a charm but some machines are still not generating any vuln report. Is there a troubleshooting how-to? I reviewed the configuration on those clients machines and apparently it looks fine. 
I got an interesting piece of info: the majority of the machines that aren't receiving any vuln scan info are all SUSE - SLES 12 -SP4, and there isn't an entry for suses in the ossec.conf in the documentation. The we have the mysteries: 2 ubuntus, 20.10 and 20.04 and a windows machine who aren't performing vuln scans, as well. 

Thanks, 
Sandro

**Reply by HanesSciarrone** [reply]
Hi @SandroCristiano 

Currently, Wazuh doesn't support SUSE distribution in the vulnerability detector this is why you don't see the configuration block. We have an epic open for adding SUSE distribution support. I leave you the [link](https://github.com/wazuh/wazuh/issues/7171) where you can follow the development if you wish.  

Regarding another OSs I would like to ask if you can put the process in debug mode and send me the log. Also, It will be very useful to have the ossec.conf also, can you send me it?

Launching the `modulesd` process in debug mode you should follow these steps:

```
kill -s SIGKILL $(pidof wazuh-modulesd)
cd /var/ossec/bin
./wazuh-modulesd -fdd
```
Options:

| Option | Description |
|:----------:|-----------------|
| -f | Run in the foreground. |
| -dd | Verbose debug mode. | 

The log is in `/var/ossec/logs/ossec.log`

Best regards
Hanes

**Reply by SandroCristiano** [reply]
Thank you! I'll do it.

**Reply by SandroCristiano** [reply]
The workaround for Suse worked as a charm. Thanks a lot! I attached the log of the ubuntu machine which the vuln scan isn't performing.
[ossec.log](https://github.com/wazuh/wazuh/files/8633696/ossec.log)

**Reply by HanesSciarrone** [reply]
Hi @SandroCristiano

This log file is of the agent right? If it is, I have seen is that you are configure the vulnerability detector in the `ossec.conf` of the agent. This should be done only in the manager. I need the manager log with wazuh-modulesd in debug mode.

Best regards
Hanes

**Reply by SandroCristiano** [reply]
Oh...my...gonna do it right now.

**Reply by SandroCristiano** [reply]
When I try to attach the file, I get the "is not in the list" message.

**Comment by SandroCristiano** [comment]
[ossec.zip](https://github.com/wazuh/wazuh/files/8635578/ossec.zip)
 K...I had to zip it.

**Comment by SandroCristiano** [comment]
Hello! I got a new doubt concerning Vuln scans. In alienvault, it was possible to run some scans on client machines without the need to install an agent. Is such a thing possible with Wazuh? Thanks again.

**Reply by HanesSciarrone** [reply]
Hi @SandroCristiano 

Sorry for my delay. Regarding this question, Vulnerability detector first looks for the connected agents in the manager and gets the ID of the agents and then looks  for the packages database collected by `syscollector` of  each agent, therefore it is necessary that you have an agent installed in the machine that you want to scan for vulnerabilities. 

Best regards
Hanes

**Reply by SandroCristiano** [reply]
Thanks a lot for your response. I asked the same question in another topic, and wasn't able to cancel it, though. Feel free to close it. Also, did you have time to check on the logs? I found out the machines who aren't collecting data from vuln scan, aren't actually communicating with the manager (netstat returns nothing). The client insists there's nothing in the network that could possibly hinder communication, said machine is actually a clone of another one which collects data normally.

**Comment by HanesSciarrone** [comment, marked as answer]
Hi @SandroCristiano 

Following with the problem that you have with some agents, the logs you sent me I could see that some agents in Ubuntu aren't supported by the Vulnerability Detector. An example of this are the agents 014 and 028.

```
2022/05/05 19:07:49 wazuh-modulesd:vulnerability-detector[4984] wm_vuln_detector.c:4936 at wm_vuldet_set_agents_info(): DEBUG: (5434): Agent '014' has an unsupported OS version: 'Ubuntu'
2022/05/05 19:07:49 wazuh-modulesd:vulnerability-detector[4984] wm_vuln_detector.c:4936 at wm_vuldet_set_agents_info(): DEBUG: (5434): Agent '028' has an unsupported OS version: 'Ubuntu'
```

That's why you don't see the vulnerabilities.

Best regards
Hanes

**Reply by SandroCristiano** [reply]
Awesome! Is there a workaround like what was done with the Suse machine in my original post, or not?

**Reply by HanesSciarrone** [reply]
Hi @SandroCristiano 

Until version 4.3 of Wazuh Ubuntu 22 wasn't supported but we are working on version 4.4 and support for this version of Ubuntu has already been added.
In addition, we have added this feature for next release https://github.com/wazuh/wazuh/issues/13162

Best regards
Hanes

**Reply by SandroCristiano** [reply]
Thank you very much for all the info and help.

Best regards,
Sandro.

**Reply by HanesSciarrone** [reply]
Hi @SandroCristiano 

Thanks to you for using and trusting Wazuh. I hope you can do everything you need with it.

Best regards
Hanes
